### PR TITLE
Googlefit integration

### PR DESCRIPTION
**Pull request by carolinux**
# Status May 7: Ready for final look
- Can connect GF through the QF website, and sync data files. Tried with own files, synced a few empty jsons since I don't have heart rate data
- Local google oauth app set up by carolinux for testing
- Prod google oauth set up by @gedankenstuecke , necessary env vars are already in heroku env
- Added a Makefile with common commands and instructions how to run locally

Points already clarified as part of this PR - left for historical reasons

- How exactly does QF get existing integrations from Open Humans? (if I've already synced GF from GoogleFit's activity page)
- Figure out what to do for `analyze_existing_reports` and `analyze_existing_events`.
```
def analyze_existing_events(user_id):
    user = User.objects.get(id=user_id)
    events = RetrospectiveEvent.objects.filter(member=user.openhumansmember)
    for event in events:
        print("Processing user {} for event {}…".format(user, event))
        analyze_event.delay(event_id=event.id)
```
Is this user id the same as the openhumansmember or another user id?

### CODE REVIEW

**Comment by gedankenstuecke** [comment]
>     * How exactly does QF get existing integrations from Open Humans? (if I've already synced GF from GoogleFit's activity page)

It depends on what you mean by _get_. [`identify_missing_sources`](https://github.com/OpenHumans/quantified-flu/blob/master/quantified_flu/helpers.py#L11) i used to check whether a person already has a given data type in their account, it does so by just iterating over all files and checking whether a given file is already in their account. If that's not the case it also checks the presence of the local `Fitbit` and `Oura` models. 


>     * Figure out what to do for `analyze_existing_reports` and `analyze_existing_events`.

Those basically just iterate over a user's `RetrospectiveEvents` and queue a celery task with `analyze_event`, in case of `analyze_existing_reports` it queues a `add_wearable_to_symptom` celery task. Both of those celery tasks are the ones where it tries to fetch the different data sources and where Google Fit would need to be added.

> Is this user id the same as the openhumansmember or another user id?

Nope, that's the internal Django app user-ID. In a view it's available via `request.user`. For a given `OpenHumansMember` it's available as `openhumansmember.user`.

File: import_data/activity_parsers.py [hunk 1]
```diff
@@ -80,6 +84,69 @@ def fitbit_parser(fitbit_info, event_start, event_end=None):
 
     return returned_fitbit_data
 
+def googlefit_to_qf(json_data, min_date, max_date):
+    res = []
+    data = json_data
+    df, _ = googlefit_parse_utils.get_dataframe_with_all(data)
+    if not df:
```
Comment (gedankenstuecke) [inline import_data/activity_parsers.py:91]: This crashed with a `ValueError`, saying that evaluating True/False for df can not be safely done as it's ambiguous. To get around this i changed it to 

```
    if isinstance(df, list):
        return res
```

instead. With that I could continue

File: import_data/activity_parsers.py [hunk 2]
```diff
@@ -80,6 +84,69 @@ def fitbit_parser(fitbit_info, event_start, event_end=None):
...
+        return res
+    # we can have multiple sources of heart rate data. will get the one with the most data per day
+    ds_with_most_data = None
+    max_data_points = 0
+    for col_name in df['columns']:
```
Comment (gedankenstuecke) [inline import_data/activity_parsers.py:96]: This one crashes for me due to the following:

```
3:28:09 PM worker.1 |  [2020-05-07 13:28:09,865: ERROR/ForkPoolWorker-1] Task retrospective.tasks.analyze_event[93bbf27e-3163-4051-94ef-9f157e7775c0] raised unexpected: KeyError('columns',)
3:28:09 PM worker.1 |  Traceback (most recent call last):
3:28:09 PM worker.1 |    File "/Users/bastian/.local/share/virtualenvs/quantified-flu-eHL-Yns6/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2646, in get_loc
3:28:09 PM worker.1 |      return self._engine.get_loc(key)
3:28:09 PM worker.1 |    File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
3:28:09 PM worker.1 |    File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
3:28:09 PM worker.1 |    File "pandas/_libs/hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
3:28:09 PM worker.1 |    File "pandas/_libs/hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
3:28:09 PM worker.1 |  KeyError: 'columns'
3:28:09 PM worker.1 |  During handling of the above exception, another exception occurred:
3:28:09 PM worker.1 |  Traceback (most recent call last):
3:28:09 PM worker.1 |    File "/Users/bastian/.local/share/virtualenvs/quantified-flu-eHL-Yns6/lib/python3.6/site-packages/celery/app/trace.py", line 385, in trace_task
3:28:09 PM worker.1 |      R = retval = fun(*args, **kwargs)
3:28:09 PM worker.1 |    File "/Users/bastian/.local/share/virtualenvs/quantified-flu-eHL-Yns6/lib/python3.6/site-packages/celery/app/trace.py", line 650, in __protected_call__
3:28:09 PM worker.1 |      return self.run(*args, **kwargs)
3:28:09 PM worker.1 |    File "/Users/bastian/dev/quantified-flu/retrospective/tasks.py", line 111, in analyze_event
3:28:09 PM worker.1 |      analyze_googlefit_event(googlefit_data, event)
3:28:09 PM worker.1 |    File "/Users/bastian/dev/quantified-flu/retrospective/tasks.py", line 88, in analyze_googlefit_event
3:28:09 PM worker.1 |      googlefit_hr_data = googlefit_parser(googlefit_data, event.date)
3:28:09 PM worker.1 |    File "/Users/bastian/dev/quantified-flu/import_data/activity_parsers.py", line 146, in googlefit_parser
3:28:09 PM worker.1 |      data_in_qf_format = googlefit_to_qf(googlefit_json, min_date, max_date)
3:28:09 PM worker.1 |    File "/Users/bastian/dev/quantified-flu/import_data/activity_parsers.py", line 98, in googlefit_to_qf
3:28:09 PM worker.1 |      for col_name in df['columns']:
3:28:09 PM worker.1 |    File "/Users/bastian/.local/share/virtualenvs/quantified-flu-eHL-Yns6/lib/python3.6/site-packages/pandas/core/frame.py", line 2800, in __getitem__
3:28:09 PM worker.1 |      indexer = self.columns.get_loc(key)
3:28:09 PM worker.1 |    File "/Users/bastian/.local/share/virtualenvs/quantified-flu-eHL-Yns6/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2648, in get_loc
3:28:09 PM worker.1 |      return self._engine.get_loc(self._maybe_cast_indexer(key))
3:28:09 PM worker.1 |    File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
3:28:09 PM worker.1 |    File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
3:28:09 PM worker.1 |    File "pandas/_libs/hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
3:28:09 PM worker.1 |    File "pandas/_libs/hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
3:28:09 PM worker.1 |  KeyError: 'columns'
```

Comment (carolinux) [reply import_data/activity_parsers.py]: yes, those were some small errors with data handling, fixed now

**Pull request merged**